### ISSUE TITLE
Empty railroad diagrams with Combine

### ISSUE THREAD

**Issue by jwhite242** [pyparsing/pyparsing#361]
So, I seem to have found a corner case in the railroad diagram creation that's triggered by specific uses of Combine() in the grammars.  If the Combine only wraps part of the grammar then the diagram looks as expected, however if the combine wraps the entire grammar, then the diagram is empty.  The snippets below demonstrate the working case and the broken case for a simple time string grammar:

* Empty railroad diagram: Combine wrapping entire grammar
  ``` python
  import pyparsing as pp
  
  ints = pp.Word(pp.nums).setParseAction(lambda toks: int(toks[0]))
  
  time_parser_1 = pp.Combine(ints('hours') + pp.Literal(":") + ints('minutes') + pp.Literal(":") + ints('seconds'))
   
  time_parser_1.create_diagram("time_parser_1.html")
  ```

* Expected railroad diagram: Combine wrapping only part of grammar
  ``` python
  import pyparsing as pp
  
  ints = pp.Word(pp.nums).setParseAction(lambda toks: int(toks[0]))
  
  time_parser_2 = pp.Combine(ints('hours') + pp.Literal(":") + ints('minutes')) + pp.Literal(":") + ints('seconds')
  
  time_parser_2.create_diagram("time_parser_2.html")
  ```
  Note the difference is the last ":" + integer part of the grammar is outside the Combine in this case

ON a side note, this is looking to be a pretty great way to document parsers and love that it's been added!

**Comment by ptmcg** [comment]
Sorry this took so long, I think I am near to a fix, to go out in 3.0.8.

**Comment by jwhite242** [comment]
Oh, no worries on the time.  Thanks for digging into it and finding a fix!

**Comment by ptmcg** [comment]
Released in 3.0.8
